filesystem/amlogic_emmc_partitions.ksy

```diff
@@ -0,0 +1,68 @@
+meta:
+  id: amlogic_emmc_partitions
+  title: Amlogic proprietary eMMC partition table
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  endian: le
+
+doc: |
+  This is an unnamed and undocumented partition table format implemented by
+  the bootloader and kernel that Amlogic provides for their Linux SoCs (Meson
+  series at least, and probably others). They appear to use this rather than GPT,
+  the industry standard, because their BootROM loads and executes the next stage
+  loader from offset 512 (0x200) on the eMMC, which is exactly where the GPT
+  header would have to start. So instead of changing their BootROM, Amlogic
+  devised this partition table, which lives at an offset of 36MiB (0x240_0000)
+  on the eMMC and so doesn't conflict. This parser expects as input just the
+  partition table from that offset. The maximum number of partitions in a table
+  is 32, which corresponds to a maximum table size of 1304 bytes (0x518).
+  
+doc-ref:
+  - https://aml-code.amlogic.com/kernel/common.git/tree/include/linux/mmc/emmc_partitions.h?id=18a4a87072ababf76ea08c8539e939b5b8a440ef
+  - https://aml-code.amlogic.com/kernel/common.git/tree/drivers/amlogic/mmc/emmc_partitions.c?id=18a4a87072ababf76ea08c8539e939b5b8a440ef
+
+seq:
+  - id: magic
+    contents: "MPT\0"
```

Comment:
I'd go for the mixed [`contents`](https://doc.kaitai.io/user_guide.html#magic) syntax here:

```suggestion
    contents: ["MPT", 0]
```

While the string escape works too, it's generally a better idea to include bytes as integers in the list.